### ISSUE TITLE
`TensorFlowModelDataSet` broken with `fsspec>=2023.3.0`

### ISSUE THREAD

**Issue by astrojuanlu** [kedro-org/kedro-plugins#359]
## Description
As per title

## Steps to Reproduce
```python
from kedro_datasets.tensorflow import TensorFlowModelDataSet
import tensorflow as tf
import numpy as np

data_set = TensorFlowModelDataSet("data/06_models/tensorflow_model.h5")

inputs = tf.keras.Input(shape=(3,))
x = tf.keras.layers.Dense(4, activation=tf.nn.relu)(inputs)
outputs = tf.keras.layers.Dense(5, activation=tf.nn.softmax)(x)
model = tf.keras.Model(inputs=inputs, outputs=outputs)

input_data = np.array([[0.5, 0.3, 0.2]])
predictions = model.predict(input_data)
data_set.save(model)

loaded_model = data_set.load()
input_data2 = np.array([[0.8, 0.4, 0.8]])
new_predictions = loaded_model.predict(input_data2)
np.testing.assert_allclose(predictions, new_predictions, rtol=1e-6, atol=1e-6)
```

(credit to @merelcht for the reproducer script)

## Expected Result
With `fsspec==2023.1.0`:

```
1/1 [==============================] - 0s 33ms/step
WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
1/1 [==============================] - 0s 15ms/step
Traceback (most recent call last):
  File "/Users/juan_cano/Projects/QuantumBlack Labs/tmp/tftest.py", line 19, in <module>
    np.testing.assert_allclose(predictions, new_predictions, rtol=1e-6, atol=1e-6)
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 1504, in assert_allclose
    assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/numpy/testing/_private/utils.py", line 797, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-06, atol=1e-06

Mismatched elements: 5 / 5 (100%)
Max absolute difference: 0.11386658
Max relative difference: 0.72159785
 x: array([[0.230395, 0.162806, 0.16625 , 0.195174, 0.245375]], dtype=float32)
 y: array([[0.296278, 0.094997, 0.096567, 0.152916, 0.359242]], dtype=float32)
```

## Actual Result
With `fsspec>=2023.3.0`:

```
1/1 [==============================] - 0s 30ms/step
WARNING:tensorflow:Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
Traceback (most recent call last):
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/kedro/io/core.py", line 192, in load
    return self._load()
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/kedro_datasets/tensorflow/tensorflow_model_dataset.py", line 153, in _load
    model = tf.keras.models.load_model(path, **self._load_args)
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/keras/src/saving/saving_api.py", line 262, in load_model
    return legacy_sm_saving_lib.load_model(
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/tensorflow/python/saved_model/loader_impl.py", line 119, in parse_saved_model
    raise IOError(
OSError: SavedModel file does not exist at: /var/folders/r7/ywj0_kvj0mxfkdkx0jrgh73r0000gn/T/kedro_tensorflow_tmpp3rqwbyu/{saved_model.pbtxt|saved_model.pb}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/juan_cano/Projects/QuantumBlack Labs/tmp/tftest.py", line 16, in <module>
    loaded_model = data_set.load()
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/kedro/io/core.py", line 613, in load
    return super().load()
  File "/Users/juan_cano/.micromamba/envs/tftest310/lib/python3.10/site-packages/kedro/io/core.py", line 201, in load
    raise DatasetError(message) from exc
kedro.io.core.DatasetError: Failed while loading data from data set TensorFlowModelDataSet(filepath=data/06_models/tensorflow_model.h5, load_args={}, protocol=file, save_args={'save_format': tf}).
SavedModel file does not exist at: /var/folders/r7/ywj0_kvj0mxfkdkx0jrgh73r0000gn/T/kedro_tensorflow_tmpp3rqwbyu/{saved_model.pbtxt|saved_model.pb}
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used (`pip show kedro` or `kedro -V`): 0.18.13
* Kedro plugin and kedro plugin version used (`pip show kedro-airflow`): 1.7.0
* Python version used (`python -V`): 3.10.12
* Operating system and version: macOS Ventura 

cc @RyanSaxe 

more discussion https://linen-slack.kedro.org/t/15730998/hey-in-working-with-kedro-on-a-macbook-m1-i-came-across-an-i#0524340d-7259-4466-8b69-891a6995d918

**Comment by felipemonroy** [comment]
Same problem here:

- kedro 0.18.13
- kedro-datasets 1.7.0
- tensorflow: 2.13.0
- python version: 3.11.5
- operation system: Ubuntu 22.04.3 LTS

**Comment by astrojuanlu** [comment]
Looks like the fix is trivial: https://github.com/kedro-org/kedro-plugins/issues/359

**Comment by merelcht** [comment]
Looks like this is the same as https://github.com/kedro-org/kedro-plugins/issues/325 and both are now fixed by #410?
